diff --git a/CMakeLists.txt b/CMakeLists.txt
index b56953c..d2baccf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,7 @@ option(MI_DEBUG_UBSAN       "Build with undefined-behavior sanitizer (needs clan
 option(MI_CHECK_FULL        "Use full internal invariant checking in DEBUG mode (deprecated, use MI_DEBUG_FULL instead)" OFF)
 option(MI_INSTALL_TOPLEVEL  "Install directly into $CMAKE_INSTALL_PREFIX instead of PREFIX/lib/mimalloc-version" OFF)
 
+include(GNUInstallDirs)
 include("cmake/mimalloc-config-version.cmake")
 
 set(mi_sources
@@ -207,14 +208,12 @@ endif()
 # Install and output names
 # -----------------------------------------------------------------------------
 
-if (MI_INSTALL_TOPLEVEL)
-  set(mi_install_libdir   "lib")
-  set(mi_install_incdir   "include")
-  set(mi_install_cmakedir "cmake")
-else()
-  set(mi_install_libdir   "lib/mimalloc-${mi_version}")
-  set(mi_install_incdir   "include/mimalloc-${mi_version}")
-  set(mi_install_cmakedir "share/mimalloc-${mi_version}/cmake") 
+set(mi_install_libdir   "${CMAKE_INSTALL_LIBDIR}")
+set(mi_install_incdir   "${CMAKE_INSTALL_INCLUDEDIR}")
+set(mi_install_cmakedir "${CMAKE_INSTALL_LIBDIR}/cmake/mimalloc-${mi_version}")
+if (NOT MI_INSTALL_TOPLEVEL)
+  set(mi_install_libdir   "${mi_install_libdir}/mimalloc-${mi_version}")
+  set(mi_install_incdir   "${mi_install_incdir}/mimalloc-${mi_version}")
 endif()
 
 if(MI_SECURE)
